src/main/java/com/microsoft/graph/tasks/PageIterator.java

```diff
@@ -0,0 +1,494 @@
+     * @return a CompletableFuture that completes when the iteration is complete.
+     * @throws ServiceException if the request was unable to complete for any reason.
+     * @throws ReflectiveOperationException if the entity or collection page could not be instantiated or if they are of invalid types.
+     */
+    @Nonnull
+    public CompletableFuture<Void> iterate() throws ServiceException, ReflectiveOperationException {
+        if(this.state == PageIteratorState.DELTA) {
+            interpageIterate().join();
+        }
+        boolean shouldContinueInterpageIteration = intrapageIterate().join();
+        while (shouldContinueInterpageIteration) {
+            interpageIterate().join();
+            shouldContinueInterpageIteration = intrapageIterate().join();
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    /**
+     * Resumes the iteration over the collection of entities in the collation page.
+     * @return a CompletableFuture that completes when the iteration is complete.
+     * @throws ServiceException if the request was unable to complete for any reason.
+     * @throws ReflectiveOperationException if the entity or collection page could not be instantiated or if they are of invalid types.
+     */
+    @Nonnull
+    public CompletableFuture<Void> resume() throws ServiceException, ReflectiveOperationException {
+        return CompletableFuture.completedFuture(iterate().join());
+    }
+
+    /**
+     * Extracts the list of entities from the Parsable collection page.
+     * @param parsableCollection the Parsable collection page.
+     * @return the list of entities.
+     * @param <TEntity> the type of the entity.
+     * @param <TCollectionPage> the type of the collection page.
+     * @throws IllegalAccessException if the Parsable does not contain a collection property.
+     * @throws InvocationTargetException if the Parsable does not contain a collection property.
+     */
+    @Nonnull
+    @SuppressFBWarnings
+    protected static <TEntity extends Parsable, TCollectionPage extends Parsable & AdditionalDataHolder> List<TEntity> extractEntityListFromParsable(@Nonnull TCollectionPage parsableCollection) throws  IllegalAccessException, InvocationTargetException {
+        try{
+            return (List<TEntity>) parsableCollection.getClass().getDeclaredMethod("getValue").invoke(parsableCollection);
+        } catch (NoSuchMethodException e) {
+            throw new IllegalAccessException("NO_COLLECTION_PROPERTY_ERROR");
+        }
+    }
+    private static <TCollectionPage extends Parsable & AdditionalDataHolder> String extractNextLinkFromParsable(@Nonnull TCollectionPage parsableCollection, @Nullable String getNextLinkMethodName) throws ReflectiveOperationException {
+        String methodName = getNextLinkMethodName == null ? CoreConstants.CollectionResponseMethods.GET_ODATA_NEXT_LINK : getNextLinkMethodName;
+        Method[] methods = parsableCollection.getClass().getDeclaredMethods();
```

Comment:
This way if the method does not exist, we can go to the additionalData holder and check for the info there rather than catching an exception in case the method does not exist. As you mentioned earlier though, if were not expecting to catch this info in the additionalData then perhaps this wont matter since we wont be checking for it there anyways.